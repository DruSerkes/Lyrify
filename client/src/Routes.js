import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
// TODO import components for search, playing, home & include in their Routes

const Routes = () => {
	return (
		<Switch>
			<Route path="/search">
				<h1>Search</h1>
			</Route>
			<Route path="/playing">
				<h1>Playing</h1>
			</Route>
			<Route path="/">
				<h1>Home</h1>
			</Route>
			<Redirect to="/" />
		</Switch>
	);
};

export default Routes;
