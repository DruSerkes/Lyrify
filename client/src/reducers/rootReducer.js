import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
