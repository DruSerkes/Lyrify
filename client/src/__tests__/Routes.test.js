import React from 'react';
import { render } from '@testing-library/react';
import Routes from '../Routes';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
const songData = {
	id     : 1,
	song   : 'test song',
	artist : 'the testers'
};
it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Routes />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Routes />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
