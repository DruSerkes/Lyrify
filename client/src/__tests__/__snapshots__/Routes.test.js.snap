// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3 Home"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Lyrify
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorSecondary"
    >
      <em>
        lyrics for the songs you love
      </em>
    </p>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-Login MuiButton-containedPrimary"
        href="https://lyrify-server.herokuapp.com/spotify/auth"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Login with Spotify
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
